lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -56,6 +65,15 @@ defmodule Kernel.ParallelCompiler do
     spawn_compilers(files, path, options)
   end
 
+  @doc """
+  Requires the given files in parallel.
+  Read `files/2` for more information.
+  """
+  def require_files(files, options \\ []) do
```

Comment:
The parallel compiler and parallel require have very distinct semantics. The compiler is capable of solving dependency conflicts, the parallel require is not. Maybe we should be able to unify both but that is certainly out of scope of this PR.

Why are you unifying both?